In [1]:
import numpy as np
import pandas as pd
solution = pd.read_csv('finalsolution2025.csv')[['ID', 'Pred']]

# my solution
old_submission_logistic = pd.read_csv('LogisticRegressionWithElo_NEW.csv')
corrected_submission_logistic = pd.read_csv('LogisticRegressionWithElo_NEW_corrected_train_2024.csv')

# Norbert's solution on updated data
old_submission_elo = pd.read_csv('only_elo_submission.csv')
corrected_submission_elo = pd.read_csv('only_elo_submission_corrected.csv')

Logistic regression

In [2]:
# Submitted, fit on 2010-2023
df = pd.merge(solution, old_submission_logistic, on='ID', how='left')
np.mean((df['Pred_x']-df['Pred_y'])**2)

0.1465888879794976

In [3]:
# Submitted, fit on 2024
df = pd.merge(solution, corrected_submission_logistic, on='ID', how='left')
np.mean((df['Pred_x']-df['Pred_y'])**2)

0.12924049079857117

Only elo

In [4]:
# Ooutdated information
df = pd.merge(solution, old_submission_elo, on='ID', how='left')
np.mean((df['Pred_x']-df['Pred_y'])**2)

0.13594697459795352

In [5]:
# Actual information
df = pd.merge(solution, corrected_submission_elo, on='ID', how='left')
np.mean((df['Pred_x']-df['Pred_y'])**2)

0.1299269829986859

We have got 760th place with incorrect only elo sumbission.

After fix the best submission would be logistic regression with brier of 0.12924049079857117 giving us 654th place.

Only elo submission after fix scored 0.1299269829986859 giving 664th place.